pyhms/demes/single_pop_eas/multiwinner.py

```diff
@@ -0,0 +1,190 @@
+from typing import Callable, Protocol
+
+import numpy as np
+
+from ...core.population import Population
+from ...core.problem import Problem
+from .common import VariationalOperator
+
+
+class UtilityFunction:
+    """
+    Utility function for multiwinner voting.
+
+    For more details, please refer to these papers:
+    1. Faliszewski, Piotr, et al. "Multiwinner voting in genetic algorithms for
+    solving ill-posed global optimization problems."
+    """
+
+    def __init__(self, distance: Callable, gamma: Callable, delta: Callable) -> None:
+        self.distance = distance
+        self.gamma = gamma
+        self.delta = delta
+
+    def evaluate(
+        self,
+        voter_genome: np.ndarray,
+        candidate_genome: np.ndarray,
+        candidate_fitness: float,
+        problem: Problem,
+    ) -> float:
+        self.h = (lambda x: x) if problem.maximize else (lambda x: 1 / (1 + x))
+        reversed_fitness = self.h(candidate_fitness)
+        distance_value = self.distance(voter_genome, candidate_genome)
+        return self.gamma(reversed_fitness) * self.delta(distance_value)
+
+    def evaluate_population(self, population: Population) -> np.ndarray:
+        # Multiwinner behavior depends on the sign of the fitness values.
+        # In order to use 1/(1+x) as a reversal function, fitness values must be different than -1.
+        positive_fitnesses = population.fitnesses - np.min(population.fitnesses)
+        scores = []
+        for voter_genome in population.genomes:
+            candidate_scores = []
+            for candidate_genome, candidate_fitness in zip(population.genomes, positive_fitnesses):
+                # I decided to adjust original implementation.
+                # Each voter should not vote for itself.
+                # This way SNTVPolicy makes more sense.
```

Comment:
I agree that SNTVPolicy makes little sense with original selfish utility function. I'm not sure how much of an effect it has on other policies, but as for my intuition, the adjustment should preserve their effect.